lib/middleman-search/extension.rb

```diff
@@ -24,7 +24,7 @@ def search_lunr_js_pipeline
       end
 
       def search_index_path
-        "/" + sitemap.find_resource_by_path(extensions[:search].options[:index_path]).destination_path
+        app.config[:http_prefix] + sitemap.find_resource_by_path(extensions[:search].options[:index_path]).destination_path
```

Comment:
`app` isn't defined on MM3, so I was trying something like this:

```
config = app.config unless config
config[:http_prefix] + ...
```

But I'll still have to test the whole extension on a MM4 site to check - I'm having some trouble using the helpers on a `.js.erb` file there.